### PR TITLE
Moving introduction to temporal cloud under evaluate

### PR DESCRIPTION
**Pull request by jsundai**
## What does this PR do?
- moves intro to temporal cloud (overview, pricing, etc) to evaluate after features
- adds overview page into the sidebar 
- adds self-hosted vs cloud content in overview page 

## Notes to reviewers
[Preview](https://vercel.live/link/temporal-documentation-git-introcloudevaluate.preview.thundergun.io?via=deployment-domains-list-branch)
TO-DO: potentially vercel.json

### CODE REVIEW

File: docs/evaluate/temporal-cloud/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+toc_max_heading_level: 4
+keywords:
+  - explanation
+  - term
+tags:
+  - explanation
+  - term
+---
+
+[Temporal Cloud](https://temporal.io/cloud) serves as a software-as-a-service (SaaS) infrastructure platform, specifically designed to manage the durability of your Temporal Applications.
```
Comment (MasonEgger) [inline docs/evaluate/temporal-cloud/overview.mdx:16]: Do we hyphenate this?  There's a lot of different ways, and I'll admit I don't know ours. If we do, just decline this suggestion. 

```suggestion
[Temporal Cloud](https://temporal.io/cloud) serves as a software as a service (SaaS) infrastructure platform, specifically designed to manage the durability of your Temporal Applications.
```

File: docs/evaluate/temporal-cloud/overview.mdx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+### Features of Temporal Cloud
+
+Based on the open-source Temporal Server software, Temporal Cloud offers comparable features without the overhead of setting up and managing a production-level Temporal Service.
+It provides [units of isolation called Namespace](/cloud/security#namespace-isolation), which you can provision for different environments such as development, testing, integration, staging, and production.
+
+Temporal Cloud provides additional features and benefits, such as:
+
+- [Globally available](/cloud/service-availability): Temporal Cloud is compatible with applications deployed in various cloud environments or data centers and operates globally.
+- [History Export](/cloud/export): Temporal Cloud allows you to export your Workflow Execution Event Histories for long-term storage, auditing, or analysis purposes.
+- [RBAC](/cloud/saml): Temporal Cloud provides RBAC which supports SAML Single Sign-On (SSO) authentication. This provides a secure and seamless way for users to access Temporal Cloud resources using their organization's credentials, simplifying user management and enhancing security.
```
Comment (MasonEgger) [inline docs/evaluate/temporal-cloud/overview.mdx:47]: We shouldn't assume users will know that acronym. 

```suggestion
- [Role Based Access Control (RBAC)](/cloud/saml): Temporal Cloud provides RBAC which supports SAML Single Sign-On (SSO) authentication. This provides a secure and seamless way for users to access Temporal Cloud resources using their organization's credentials, simplifying user management and enhancing security.
```

File: docs/evaluate/temporal-cloud/overview.mdx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+- [Support](/cloud/support): Temporal Cloud includes the support, services and training needed to onboard you successfully, design and deploy your application efficiently and scale. Our team has extensive knowledge of the Temporal project, and a broad set of skills to help you succeed with any project.
+- [Consumption based pricing](/cloud/pricing): Temporal Cloud is a consumption-based service; you pay only for what you need when you need it. Our pricing derives from your use of actions, storage, and support. Pricing is flexible, transparent, and predictable, so you know your costs and never pay for unused capacity.
+
+### High level system topology
+
+There are two major parts of the Temporal Platform that work together to create the fully functioning system.
```
Comment (MasonEgger) [inline docs/evaluate/temporal-cloud/overview.mdx:53]: This sections seems to be generically describing Temporal, yet it lists Temporal Cloud as the second part. Should this be Temporal Service, in which we state that Temporal Cloud is an offering of the Temporal Service.

**Review by MasonEgger — CHANGES_REQUESTED** [review]

**Review by axfelix — APPROVED** [review]
Approving pending Mason's comments

**Pull request merged**